File: OGMA - Gilliardi/__projects/Coc.Modeling.Uml/UmlDiagram.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Coc.Modeling.Uml
{
    public abstract class UmlDiagram
    {
        /// <summary>
        /// Stores this diagram name
        /// </summary>
        public String Name { get; set; }

        public String Id { set; get; }

        /// <summary>
        /// Stores uml objects for this model. 
        /// TODO: See UML documentation for validation.
        /// </summary>
        public List<UmlBase> UmlObjects { get; private set; }

        /// <summary>
        /// Stores stereotype set for this model.
        /// TODO: Should be vailable by profile.
        /// </summary>
        public List<String> UmlStereotypes { get; set; }

        /// <summary>
        /// Default constructor.
        /// </summary>
        public UmlDiagram()
        {
            this.UmlObjects = new List<UmlBase>();
            this.UmlStereotypes = new List<String>();
            this.Id = Guid.NewGuid().ToString();
        }

        public UmlElement GetElementById(String id)
        {
            foreach (UmlBase element in UmlObjects.OfType<UmlBase>())
            {
                if (element.Id.Equals(id))
                {
                    return (UmlElement) element;
                }
            }
            return null;
        }

        public UmlModel ParentModel { get; set; }

        public override string ToString()
        {
            return this.Name;
        }
    }
}

File: OGMA - Gilliardi/AutoFun/__projects/ArgoParser/src/Diagrams/UmlDiagram.java
package Diagrams;

import java.util.ArrayList;
import Objects.*;
import java.util.UUID;

public abstract class UmlDiagram
{
    private String Id= UUID.randomUUID().toString();;

    private String Name;

    private UmlModel ParentModel;

    private ArrayList<UmlBase> UmlObjects = new ArrayList<>();

    private ArrayList<String> UmlStereotypes = new ArrayList<>();

    public UmlElement GetElementById(String id)
    {
        return null;
    }

    public String getId()
    {
        return Id;
    }

    public void setId(String Id)
    {
        this.Id = Id;
    }

    public String getName()
    {
        return Name;
    }

    public void setName(String Name)
    {
        this.Name = Name;
    }

    public UmlModel getParentModel()
    {
        return ParentModel;
    }

    public void setParentModel(UmlModel ParentModel)
    {
        this.ParentModel = ParentModel;
    }

    public ArrayList<UmlBase> getUmlObjects()
    {
        return UmlObjects;
    }

    public void setUmlObjects(ArrayList<UmlBase> UmlObjects)
    {
        this.UmlObjects = UmlObjects;
    }

    public ArrayList<String> getUmlStereotypes()
    {
        return UmlStereotypes;
    }

    public void setUmlStereotypes(ArrayList<String> UmlStereotypes)
    {
        this.UmlStereotypes = UmlStereotypes;
    }
    
    public void addUmlObject(UmlBase obj)
    {
        this.UmlObjects.add(obj);
    }
}
